src/edu/csus/ecs/pc2/clics/API202306/CLICSFileReference.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ser.FilterProvider;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS File Reference
+ * Contains information about a file used in the API
+ *
+ * @author John Buck
+ *
+ */
+
+@JsonFilter("rtFilter")
+public class CLICSFileReference {
+
+    @JsonProperty
+    private String href;
+
+    @JsonProperty
+    private String filename;
+
+    @JsonProperty
+    private String hash;
+
+    @JsonProperty
+    private String mime;
+
+    @JsonProperty
+    private int width;
+
+    @JsonProperty
+    private int height;
+
+    @JsonProperty
+    private String data;
```

Comment:
What is this property for?  It doesn't appear to be defined in the CLICS spec, and there doesn't appear to be any way to set its value...

Reply:
Aha.  It actually DOES appear in the CLICS spec but it is quasi undocumented (unless inclusion implies it's documented).  See [CLICS Spec JSON Property Types](https://ccs-specs.icpc.io/2023-06/contest_api#json-property-types) and scroll down to the example at the end of that section (just before **Capabilities**).

> If implementing support for uploading files pointed to by resource links, substitute the href property with a data property with a base64 encoded string of the associated file contents as the value.

Also see [The second example for POST](https://ccs-specs.icpc.io/2023-06/contest_api#examples-13)

Reply:
Hmmm.... ok, I see that there are examples containing `data` fields.  I guess this is an error/oversight in the CLICS spec page; if there's supposed to be support for a JSON property named `data` then (IMHO) it clearly should be listed in the `Properties for file reference objects:` table.  But I'm sure this isn't the only "hole" in the spec :) 

What about the question of _providing support for setting the property value_?  Seems like if we're going to include it then we ought it also include a way to set it...?